With USE=editor and gcc12:

    similar/main/piggy.cpp: In function 'void d2x::piggy_new_pigfile(char*)':
    similar/main/piggy.cpp:870:62: error: '%u' directive output may be truncated writing between 1 and 10 bytes into a region of size between 7 and 19 [-Werror=format-truncation=]

But the offending code is unused, so can just drop it.

Backport from:
https://github.com/dxx-rebirth/dxx-rebirth/commit/12cca97870efe419613462dc279963a77bc80d9a
--- a/similar/main/piggy.cpp
+++ b/similar/main/piggy.cpp
@@ -889,11 +889,8 @@ void piggy_new_pigfile(char *pigname)
 				}
 			
 				for (fnum=0;fnum<nframes; fnum++)       {
-					char tempname[20];
 					int SuperX;
 
-					snprintf(tempname, sizeof(tempname), "%s#%u", basename, fnum);
-
 					//SuperX = (GameBitmaps[i+fnum].bm_flags&BM_FLAG_SUPER_TRANSPARENT)?254:-1;
 					SuperX = (GameBitmapFlags[i+fnum]&BM_FLAG_SUPER_TRANSPARENT)?254:-1;
 					//above makes assumption that supertransparent color is 254
